connector-template/mod.rs

```diff
@@ -70,24 +94,70 @@ impl
 {
 }
 
-impl api::PaymentVoid for {{project-name | downcase | pascal_case}} {}
-
 impl
     ConnectorIntegration<
-        api::Void,
-        types::PaymentsCancelData,
+        api::Authorize,
+        types::PaymentsAuthorizeData,
         types::PaymentsResponseData,
-    > for {{project-name | downcase | pascal_case}}
-{}
+    > for {{project-name | downcase | pascal_case}} {
+    fn get_headers(&self, req: &types::PaymentsAuthorizeRouterData, connectors: &settings::Connectors,) -> CustomResult<Vec<(String, String)>,errors::ConnectorError> {
+        self.build_headers(req, connectors)
+    }
 
-impl api::ConnectorAccessToken for {{project-name | downcase | pascal_case}} {}
+    fn get_content_type(&self) -> &'static str {
+        self.common_get_content_type()
+    }
 
-impl ConnectorIntegration<api::AccessTokenAuth, types::AccessTokenRequestData, types::AccessToken>
-    for {{project-name | downcase | pascal_case}}
-{
+    fn get_url(&self, _req: &types::PaymentsAuthorizeRouterData, _connectors: &settings::Connectors,) -> CustomResult<String,errors::ConnectorError> {
+        Err(errors::ConnectorError::NotImplemented("get_url method".to_string()).into())
+    }
+
+    fn get_request_body(&self, req: &types::PaymentsAuthorizeRouterData) -> CustomResult<Option<String>,errors::ConnectorError> {
+        let {{project-name | downcase}}_req =
+            utils::Encode::<{{project-name | downcase}}::{{project-name | downcase | pascal_case}}PaymentsRequest>::convert_and_encode(req).change_context(errors::ConnectorError::RequestEncodingFailed)?;
+        Ok(Some({{project-name | downcase}}_req))
+    }
+
+    fn build_request(
+        &self,
+        req: &types::PaymentsAuthorizeRouterData,
+        connectors: &settings::Connectors,
+    ) -> CustomResult<Option<services::Request>, errors::ConnectorError> {
+        Ok(Some(
+            services::RequestBuilder::new()
+                .method(services::Method::Post)
+                .url(&types::PaymentsAuthorizeType::get_url(
+                    self, req, connectors,
+                )?)
+                .headers(types::PaymentsAuthorizeType::get_headers(
+                    self, req, connectors,
+                )?)
+                .body(types::PaymentsAuthorizeType::get_request_body(self, req)?)
+                .build(),
+        ))
+    }
+
+    fn handle_response(
+        &self,
+        data: &types::PaymentsAuthorizeRouterData,
+        res: Response,
+    ) -> CustomResult<types::PaymentsAuthorizeRouterData,errors::ConnectorError> {
+        logger::debug!({{project-name | downcase}}_payments_authorize_response=?res);
```

Comment:
There is a PR by @Gnanasundari24 which will add logs at a higher level. I think this won't be necessary #624